### PR TITLE
[AIRFLOW-353]when multiple tasks gets removed update status fails

### PR DESCRIPTION
**Pull request by yiqingj**
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- _https://issues.apache.org/jira/browse/AIRFLOW-353_

Testing Done:
- None,  no new features were added

### CODE REVIEW

**Comment by skudriashev** [comment]
@yiqingj please, follow the 50/72 commit message rules. Thanks.

File: airflow/models.py [hunk 1]
```diff
@@ -3536,7 +3536,7 @@ def update_state(self, session=None):
         logging.info("Updating state for {} considering {} task(s)"
                      .format(self, len(tis)))
 
-        for ti in tis:
+        for ti in tis[:]
```
Comment (skudriashev) [inline airflow/models.py]: You may want:

``` python
    for ti in tis[:]:
        ...
```

**Comment by jlowin** [comment]
I'm surprised the original code worked at all -- doesn't Python complain if an iterator changes size during iteration? cc @bolkedebruin 

@yiqingj my preference would be to say `list(tis)` rather than `tis[:]` since it's more clear that it's making a copy.

**Comment by bolkedebruin** [comment]
@yiqingj good catch this! @jlowin just tried in a notebook, no python doesnt complain (2.7). WIll merge

**Pull request closed without merging**